internal/servers/worker/handler.go

```diff
@@ -52,11 +53,26 @@ func (w *Worker) handleProxy(listenerCfg *listenerutil.ListenerConfig) (http.Han
 	return func(wr http.ResponseWriter, r *http.Request) {
```

Comment:
I know writing handler tests is a pain, but should we add at least one that verifies the error handling?